packages/formspree-react/src/context.tsx

```diff
@@ -1,98 +1,75 @@
+import { Elements } from '@stripe/react-stripe-js';
 import { loadStripe } from '@stripe/stripe-js/pure.js';
-import type { Stripe } from '@stripe/stripe-js';
-
-const Elements = lazy(() =>
-  import('@stripe/react-stripe-js').then((module) => {
-    return { default: module.Elements };
-  })
-);
-
-export type FromspreeContextType = {
+import React, {
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+  type ReactNode,
+} from 'react';
+
+export type FormspreeContextType = {
   client: Client;
 };
 
 export type FormspreeProviderProps = {
+  children: ReactNode;
   project?: string;
-  children: React.ReactNode;
   stripePK?: string;
 };
 
-const FormspreeContext = React.createContext<FromspreeContextType | null>(null);
-
-FormspreeContext.displayName = 'Formspree';
-
-let stripePromise: Promise<Stripe | null>;
-
-const getStripe = (stripeKey: string) => {
-  if (!stripePromise) {
-    stripePromise = loadStripe(stripeKey);
-  }
-  return stripePromise;
-};
-
-const handleCreateClient = (promise?: Stripe, project?: string) => {
-  const config: Config = {};
-
-  if (promise) {
-    config.stripePromise = promise;
-  }
-
-  if (project) {
-    config.project = project;
-  }
-
-  return createClient(config);
-};
-
-export const FormspreeProvider = (props: FormspreeProviderProps) => {
-  const [stateStripePromise, setStateStripePromise] = useState<
-    Stripe | undefined
-  >(undefined);
-  const [client, setClient] = useState<Client>(
-    handleCreateClient(stateStripePromise, props.project)
+const FormspreeContext = React.createContext<FormspreeContextType | null>(null);
+
+/**
+ * FormspreeProvider creates Formspree Client based on the given props
+ * and makes the client available via context.
+ */
+export function FormspreeProvider(props: FormspreeProviderProps) {
+  const { children, project, stripePK } = props;
+  const [client, setClient] = useState(createClient({ project }));
+  const stripePromise = useMemo(
+    () => (stripePK ? loadStripe(stripePK) : null),
+    [stripePK]
   );
 
   useEffect(() => {
-    const getStripePromise = async (stripeKey: string) => {
-      const promiseStripe = await getStripe(stripeKey);
-      if (promiseStripe) {
-        setStateStripePromise(promiseStripe);
-      }
-    };
-
-    if (props.stripePK) {
-      getStripePromise(props.stripePK);
+    let isMounted = true;
+    if (isMounted) {
+      setClient((client) =>
+        client.project !== project
+          ? createClient({ ...client, project })
+          : client
+      );
     }
-  }, [props.stripePK]);
+    return () => {
+      isMounted = false;
+    };
+  }, [project]);
 
   useEffect(() => {
-    if (stateStripePromise) {
-      setClient(handleCreateClient(stateStripePromise, props.project));
-    }
-  }, [props.project, stateStripePromise]);
+    let isMounted = true;
+    stripePromise?.then((stripe) => {
+      if (isMounted && stripe) {
+        setClient((client) => createClient({ ...client, stripe }));
```

Comment:
In the future, we could probably just rely on `Elements` without needing to update `client.stripe` but it'll introduce a breaking change i.e. it'll require `FormspreeProvider` to be rendered. I think we can revisit this when we extract Stripe into a separate dependency.